model/modelpb/apmevent.pb.json.go

```diff
@@ -71,33 +66,33 @@ func (e *APMEvent) toModelJSON(out *modeljson.Document) {
 		}
 	}
 
-	var transaction modeljson.Transaction
 	if e.Transaction != nil {
+		var transaction modeljson.Transaction
```

Comment:
Thank you for the feedback! 

I added benchmarks and there are quite a few allocations. I spent quite a while trying to fight the escape analysis but without much success. I'm trying to evalute how feasible it is to put together some code generation to automate the mapping.

Reply:
Thank you for the feedback!

I added benchmarks and there are quite a few allocations. I spent quite a while trying to fight the escape analysis but without much success. I'm trying to evalute how feasible it is to put together some code generation to automate the mapping but I'm afraid there's not much we can do to reduce allocation.

From what I could see from the output of escape analysis (`-m -m -m`) the variables are escaping to the heap because the struct field is a pointer (as opposed to `model.APMEvent` being composed of struct for the most part).
I tried playing around with the code but the benchmark is always reporting ~104 allocations.

Reply:
I tried another way that eliminates indirectly assigning the pointer and pre-assign them on the parent function so that they won't escape to the heap.
Allocations are down to ~59 and the mapping is definitely faster. Let me know what you think!

Note: tests are broken.

Reply:
> I tried another way that eliminates indirectly assigning the pointer and pre-assign them on the parent function so that they won't escape to the heap.
> Allocations are down to ~59 and the mapping is definitely faster. Let me know what you think!

Preassigning them doesn't seem ideal to me. Does the approach in `model.APMEvent.MarshalFastJSON` not work? The idea there is that we define the structs at the top level of the method, so they shouldn't escape.

Reply:
I've added a commit that changes back to the old approach (it can be reverted easily for testing).

Allocations went from 59 to 71 and cpu time was slightly slower.

I also added a benchmark for comparison with `model.APMEvent`.

Reply:
If the tradeoff is acceptable (increased allocations) we can keep it this way and try to improve later. I don't have a strong opinion on it. Let me know what you think